agent/discovery_chain_endpoint.go

```diff
@@ -102,13 +101,10 @@ type discoveryChainReadResponse struct {
 }
 
 func decodeDiscoveryChainReadRequest(raw map[string]interface{}) (*discoveryChainReadRequest, error) {
-	// lib.TranslateKeys doesn't understand []map[string]interface{} so we have
-	// to do this part first.
-	raw = lib.PatchSliceOfMaps(raw, nil, nil)
-
 	var apiReq discoveryChainReadRequest
 	decodeConf := &mapstructure.DecoderConfig{
 		DecodeHook: mapstructure.ComposeDecodeHookFunc(
+			decode.HookWeakDecodeFromSlice,
 			decode.HookTranslateKeys,
```

Comment:
The only caller to this function seems to only accept JSON. I suspect `PatchSliceOfMaps` was being called here unnecessarily. I'm thinking `decode.HookWeakDecodeFromSlice` can be removed.

Reply:
I wonder if it was added because of the interaction with the CLI command. The CLI command attempts to parse HCL files but IIRC with no schema which could introduce the arrays into the JSON that it then sends to the API endpoint. So the HCL parsing never occurs in this code path, but it does in the CLI that consumes this API?

Reply:
I haven't been able to find any CLI command which uses this endpoint. `DiscoveryChain.Get` exists in the `api` package, but it is not called.

Reply:
It was added in #6248, and the docs (https://www.consul.io/api-docs/discovery-chain) suggest it is not used, but exists for possible future integrations.

I suspect the PatchSliceOfMaps was copied from other code, and we don't need it. I'll add a comment about how it is likely not necessary.

Reply:
My mistake - I should have looked at more context. It's only the "PUT" side of the disco chain endpoints that my comments apply to.